homeassistant/bootstrap.py

```diff
@@ -549,7 +550,7 @@ def enable_logging(hass: core.HomeAssistant, verbose: bool=False,
             logging.Formatter('%(asctime)s %(name)s: %(message)s',
                               datefmt='%y-%m-%d %H:%M:%S'))
         logger = logging.getLogger('')
-        logger.addHandler(err_handler)
+        logger.addHandler(AsyncHandler(hass.loop, err_handler))
```

Comment:
Add a listen_once for HOMEASSISTANT_STOP here that stops and removes the handler.